### PR TITLE
Unit tests for Path2D class

### PR DESCRIPTION
**Pull request by jbcolli2**
Unit tests for the `Path2D` class.  This follows pretty closely to the Path3D tests that already exist, testing to be sure the path is initialized correctly and that the `Curve` of the path is set correctly under various conditions.

### CODE REVIEW

**Comment by MewPurPur** [comment]
Have you seen #66425?

**Comment by jbcolli2** [comment]
Nope, thanks for pointing it out.  I was working on this off of issue #43440 and didn't see anyone mention Path2d since 2020 so I thought I'd take care of it.  I'm new to contributing so not sure what to do at this point.

**Comment by jbcolli2** [comment]
Updated PR with tests from PR #66425 authored by @MatthewTave.

**Comment by Geometror** [comment]
Thanks for updating your PR so quickly! There are just two things left to do: Please squash your commits into a single one as described [here](https://docs.godotengine.org/en/stable/community/contributing/pr_workflow.html). Furthermore, please don't mention a PR in your commit message since that thread will be spammed whenever you forcepush. You can add @MatthewTave as a co-author to your single commit by following the steps [here](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors).

**Comment by jbcolli2** [comment]
Ok, I think that should do it.  Took me a sec to figure out the co author syntax.  It actually wasn't the empty line, the email has to be within brackets, which isn't shown in the official github co-author help site.  Thanks for the help.

File: tests/scene/test_path_2d.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	SUBCASE("Curve rewrite testing") {
+		const Ref<Curve2D> &curve1 = memnew(Curve2D);
+		const Ref<Curve2D> &curve2 = memnew(Curve2D);
+
+		test_path->set_curve(curve1);
+		test_path->set_curve(curve2);
+		CHECK_MESSAGE(test_path->get_curve() != curve1,
+				"After rewrite, second curve should be in class");
+		CHECK_MESSAGE(test_path->get_curve() == curve2,
+				"After rewrite, second curve should be in class");
```
Comment (Geometror) [inline tests/scene/test_path_2d.h:71]: This wordings seems a bit off to me, but that's also used in the Path3D tests, so I think we can change that later.

**Review by Geometror — APPROVED** [review]
LGTM overall, there are still test cases missing which actually test the functionality of Path2D, but as a start I think this is fine.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**